utils/wesnoth-map-diff/src/images.ts

```diff
@@ -22,6 +22,9 @@ const getDictTerrainType2ImagesPath = async () => {
       }
 
       acc[terrainType.string.value] = terrainType.symbol_image.value
+      if (!acc[terrainType.string.value]) {
```

Comment:
It isn't enough. On line 20, we have an early return when `symbol_image` is `undefined`.

We should change the logic to something like this:
```ts
.reduce((acc, terrainType) => {
  if (terrainType.symbol_image) {
    acc[terrainType.string.value] = terrainType.symbol_image.value
  }

  if (terrainType.string) {
    acc[terrainType.string.value] = terrainType.string.value
  }

  return acc
```

Reply:
so, like in [8961e19](https://github.com/wesnoth/wesnoth/pull/7452/commits/8961e19845a7c863f6cce77157df2a39cf07ff77)?

Reply:
Yeah, but with a minor change:
```diff
- if (!acc[terrainType.string.value] && terrainType.string.value) {
+ if (!acc[terrainType.string.value] && terrainType.string) {
```